src/forms/FormFieldAdvancedSelection/AdvancedSelectionDrawer.tsx

```diff
@@ -105,6 +109,10 @@ const AdvancedSelectionDrawer = (props: AdvanceSelectionDrawerPropTypes): ReactE
 			setInternalOptions();
 		}
 
+		if (fieldDef.inputSettings?.selectLimit === 0) {

```

Comment:
I think this might not be needed if we update the function from above to re-execute with a useCallback with a dependency of selectLimit